src/wolfsftp.c

```diff
@@ -3887,41 +3826,26 @@ int wolfSSH_SFTP_RecvRename(WOLFSSH* ssh, int reqId, byte* data, word32 maxSz)
     if (sz > maxSz - idx) {
         ret = WS_BUFFER_E;
     }
-    old = (char*)WMALLOC(sz + 1, ssh->ctx->heap, DYNTYPE_BUFFER);
-    if (old == NULL) {
-        ret = WS_MEMORY_E;
-    }
-    if (ret == WS_SUCCESS) {
-        WMEMCPY(old, data + idx, sz); idx += sz;
-        old[sz] = '\0';
+    if (GetAndCleanPath(ssh->sftpDefaultPath,
+                data + idx, sz, old, sizeof(old)) < 0) {
+        ret = WS_FATAL_ERROR;
```

Comment:
Does the stuff below need to be guarded by ret == WS_SUCCESS in the event we hit this error?

Reply:
Yes it should be. And I see another pattern that needs addressing related to this.